libs/ActivityKit/source/activities/ShapeRecognition.cpp

```diff
@@ -0,0 +1,54 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+// LCOV_EXCL_START
+
+#include "ShapeRecognition.h"
+#include <random>
+
+namespace leka::activity {
+
+void ShapeRecognition::start()
+{
+	_score		   = 0;
+	_current_shape = {};
+
+	_backup_callback = _rfidkit.getCallback();
+	std::shuffle(_shapes.begin(), _shapes.end(), std::mt19937(time(nullptr)));
+	launchNextRound();
+
+	auto on_tag_detected_callback = [this](const MagicCard &card) {
+		if (card == _current_shape->card) {
+			_reinforcerkit.playDefault();
+			++_score;
+
+			if (_score == kRoundsNumber) {
+				_backup_callback(MagicCard::dice_roll);
+				return;
+			}
+
+			launchNextRound();
+		} else {
+			_backup_callback(card);
+		}
+	};
```

Comment:
I have a question, why this lambda isn't a private method?

Reply:
About `lambda` vs `std::bind` [[Link](https://stackoverflow.com/questions/13307090/do-stdfunction-and-stdbind-do-dynamic-memory-allocation)]